app/models/in_person_enrollment.rb

```diff
@@ -0,0 +1,22 @@
+class InPersonEnrollment < ApplicationRecord
+  belongs_to :user
+  belongs_to :profile
+  enum status: {
+    pending: 0,
+    passed: 1,
+    failed: 2,
+    expired: 3,
+    canceled: 4,
+  }
+
+  validate :profile_belongs_to_user
+
+  private
+
+  def profile_belongs_to_user
+    unless profile&.user == user
```

Comment:
Thanks for clarifying, @tomas-nava . That makes sense. I forgot that validations can't assume that other validations has already been run, so I guess I assumed (wrongly) it would never reach this point with a `nil` `profile`.

Reply:
If we augmented the database uniqueness constraint with a Rails presence validation, and ordered it first in the class definition, we could rely on that failing before it made it to `profile_belongs_to_user`, but it doesn't look like that redundancy is common practice on this repo.